Sources/PromotedCore/MetricsLogger.swift

```diff
@@ -144,13 +105,71 @@ public final class MetricsLogger: NSObject {
     super.init()
     self.monitor.addOperationMonitorListener(self)
   }
+}
+
+// MARK: - Ancestor IDs
+public extension MetricsLogger {
+  /// Log user ID for this session. Updated when
+  /// `startSession(userID:)` or `startSessionSignedOut()` is
+  /// called.
+  var logUserID: String? {
+    get { logUserIDProducer.currentValue }
+    set { logUserIDProducer.currentValue = newValue }
+  }
 
-  // MARK: - Starting new sessions
+  /// Log user ID for this session.
+  /// If read before the first call to `startSession*`,
+  /// returns the cached ID from the previous session from
+  /// `PeristentStore`.
+  var currentOrPendingLogUserID: String? {
```

Comment:
nit - Why 'pending' vs 'next'?

Reply:
nit - I kinda view 'current' ancestor ID as whatever would be used if a log call is made.  I'd assume 'current' would be filled in correctly from previous sessions.  Maybe I'm thinking about this wrong?

Reply:
Sorry, I didn't see these comments. I agree that the names aren't great. We originally had `current` because you could be getting the current value from here, in case you read the value after the ancestor event was logged.

Open to suggestions. :-) I've also considered `peekLogUserID`.